### ISSUE TITLE
New Job: Angler - Concept and Feedback

### ISSUE THREAD

**Issue by GreatBigFailure** [d3athrow/vgstation13#27105]
Angler has been in the works for a while now and has enough content to form a solid vision. This issue is just a way to discuss the idea, get feedback/criticism, and possibly show progress as it happens. 

This is also going to be very long so that I can stop writing a novel every time someone gives me an opportunity to gush about it. 

### **Angler Concept**
Angler is a new job semi-similar to shaft miner. It will be a supply department job focused on, of course, space fishing. 

**How it's Unique**
Originally angler was going to be a shaft miner sub-job, and depending on feedback still may be, but I've been designing it around it being its own job. 
Angler is different from miner in that rather than being really good at getting one resource: materials, it is a jack-of-all-trades supply job that will be worse at getting materials than miner, but capable of supplying multiple departments with things they need. They'll still be able to get materials, and should be able to supply science with enough to meet their needs, but not enough to line the floors of the department with stacks of gold and uranium like miners can.

**But why?**
I don't like how shaft miner is necessary for science and mechanics to function while also being, in my opinion, against the spirit of spess due to it mostly being a single player game. Miners don't interact with other crew much until they deliver to science or retire. Angler aims to resolve this by constantly promoting cross-department interaction and usefulness. While anglers will spend a good amount of time off-station they'll also want to return constantly and eventually be able to do their job while still on station. 
The jack-of-all-trades concept also allows for angler to fix issues with other jobs, for example I see Xenobiology's reliance on RNG as bad design, angler will supply something to help with it. This will also, hopefully, reduce the "jobbie" mentality by making cross-department interaction more useful. Jobbies get tools, whole station gets more RP/chatter. 

### **The Job**
Angler fishes up mobs, items, and salvage from space. These are separated into conceptually different loot-tables that each attempt to do something different. 

### **Angler Mobs**
Mobs will be the most interesting and unique part of the job. They are also the most code/sprite intensive, and thankfully the closest to being complete.
Mob catches make butchering a central mechanic of angler as each one attempts to provide a use both as a living mob, and once killed. This is so that anglers will have a reason to both kill and keep them around, but also so that the whole station doesn't slowly become aquarium themed with living fish wandering around.

Sprite work has been done by SageOfLight. The man does great sprites and really, really like fish. Very lucky coincidence. He's in charge of the visual direction. 

![stonegulper](https://user-images.githubusercontent.com/64218965/87484685-ca095700-c604-11ea-811f-8aea0589c454.png)

Stonegulper, angler's primary source of ore.

![puffbumper_FINAL](https://user-images.githubusercontent.com/64218965/87484560-7c8cea00-c604-11ea-954d-9c3c14834e2e.png)

The Puffbumper

![suckerFINAL](https://user-images.githubusercontent.com/64218965/87484566-80207100-c604-11ea-9cb9-948de26aaa7d.png)

The Stationsucker

**_Example: Meel_**

Meels are intended to be used by the chef/botany. Female meels lay eggs somewhat quickly and provide less meat when butchered than males. Butchered meels produce a random type of meat (with a very small blacklist for sanity). Female meels eventually stop producing eggs when they lay too many, however if a male meel is around when they lay that egg has a small chance to produce another meel. 
So you have a reason to keep them around as a living farm animal to produce caviar. Even the males which are designed with butchering in mind are useful to keep more meels coming, and females will eventually only be useful when butchered too. 

_**Taming**_

Most angler mobs will be able to be tamed. Depending on the type of mob this might be as minor as it just not attacking you anymore, or as major as it picking up items you point to and bringing them to you. Some mobs will only provide their "living use" while tamed. Each mob has a different item required to tame it, like meat foods for space sharks or alcohol for the "Barrelshell Fermurtle" 
Most mobs won't act as pets and instead it will mostly just be a way to turn a hostile mob into a non-hostile tool. For example the Eswordfish, instead of attacking anything it sees, will attack anything except you, which will make killing some mobs for butcher a lot less dangerous. 

_**Catch Size and Mutations**_

Catch size is similar to botany's potency in function. Bigger catch (usually) means more useful fish. Meels will provide more meat. Eswordfish's obvious butcher product of a "fish-e-sword" will be stronger based on size, etc. Catch size ranges differ between fish type. A random number within the range is chosen upon catch, but things like mutations and bait can increase it. 

Mutations are intended to keep a modest pool of potential mob catches interesting and varied. They're separated by rarity and not every fish can have every mutation. Bait will increase chances of rare mutations. 
Common mutations are generally mundane things like changing numbers like size, health, damage, or the poison mutation making their attacks inject a small amount of toxin. The greedy mutation remove the mob's normal tame item and replaces it with 100 credits, etc. 
Uncommon add a use or danger to the mob like having it release a small EMP burst when angered, or just glow in the dark.
Rare mutations generally turn the mob into a sort of living large artifact from xenoarcheology. Things like causing a small gravity well for a brief moment, flinging everything around it toward the fish. Very rarely splitting into another fish of the same species, etc. These will of course be very rare and more limited to type of mob. 

_Unique mutation interactions_
While mutations add variety to mobs they primarily aim to separate their usefulness depending on mutation. Let's go back to the meel example:
Meels with the previously mentioned poison mutation will have a bit of toxin in their eggs. The glowing mutation adds a bit of luminol, clown mutation usually produces banana juice with a small chance of honk serum, etc. Royal and splitting mutated meels will produce more eggs and their eggs will have a higher chance to hatch. Illusionary meels sometimes lay fake eggs that disappear after a few seconds. 
This isn't unique to meels, most mobs have mutation-specific mechanics like this. 

### **Item Fish**

Thematically they'll still be living fish but won't actually be mobs. They'll have more singular, simple uses. They of course won't have mutations and will be caught in more abundance, and usually be consumed on use. The list is still being made but some examples:

_Space leech:_ Sucks chemicals out of its host, if they have no chemicals in them it'll suck blood. Meant for medical and a high-risk chloral counter. 

_Babelfish:_ Works like an implant. Translates languages but doesn't allow you to speak them. Might have some disadvantage like increased nutrition consumption, also rare. 

_Coinfish:_ Re-usable coin. Just swims on out of the machine after a while. 

### **Salvage**

Mostly recycled items that already exist. Corpses with a small bit of items, like a dead miner with ore or an assistant with glubbs. Stock parts, beakers, etc. 
There will also be some angler unique items like messages in bottles telling the location of vaults, or a vending machine supply pack with very expensive items akin to a mini-spessmart. 

### **How Do I Fish?**

The general flow of fishing will consist of throwing your line into space, having it drift, then "reeling in" once you get a bite. Both for theme and because it's much, much easier to code the rods will use wormhole (upgrades will use bluespace) technology to teleport the catch back to you when you reel in. 

Bait and Z level of your hook decide what can be caught.
Anglers will thus have an inherent reason to go to different Z levels, not just 1 or 5. I anticipate it becoming to go-to vault hunter job. 
Bait will vary from item-fish, mob-fish meat, and other items being usable as bait. With that said the best bait will be obtained from the chef, botany, and bar as cross-department interaction is again a central focus. 

**Scanner and Catch Cage**

A reeled in mob-fish will first be trapped in a net. Larger (usually more dangerous) fish will break out somewhat quickly. Prior to upgrades you'll want to scan the fish, telling you its species, size, and if a mutation (but not what mutation) is present. You can then transfer the mob directly from the net to a catch-cage. The cage will analyze the catch over time and tell you its mutation, you can also give it tame items while it's confined (at a lower chance to tame). Catches will (probably) have a very small chance of breaking out of the cage based on their size. 

**Harpoon and Fillet Knife**

Killing your catches will be a big, and dangerous, part of angler. While miners can choose to risk fighting mobs for rewards anglers generally don't get the choice, it's their job. I like the idea of miner's PKAs: A gun not designed to hurt the crew but rather only work in the conditions their job sees. Anglers have their own versions, a harpoon and fillet knife.
The harpoon is essentially a hookshot with much shorter range, a recharge period, and moderately high damage. When fired at a simple mob it will stun them for a significant amount of time and, like a hookshot, drag them to you. The fillet knife will do more damage to space fish as it gets by their scales. So anglers will fire their harpoon, hopefully not miss, and then finish the helpless mob with their knife. This obviously would be terrible against a human opponent. The fillet knife will also have very high butcher speed. 

**Fishing Hole**

An idea stolen from a previous fishing PR. 
High end fishing research, but not hard to make. This creates a machine that produces a wormhole you can cast your line into and reel in from. The machine can be set to allow fishing in any Z level. So once an angler has gone exploring and analyzed enough fish he can retire to the bar for comfy RP and drinks without giving up on his catches. Hopefully a ferocious Radioctive Space Shark doesn't manage to wrestle loose from his net, though. 

### **tl;dr**

Fish are cool. You're gonna chat with the chef. There's a shark. 

**Please Help I'm Out of Ideas**

I appreciate any suggestions for any element of the job, whether it's a mob, a mutation, salvage, or angler equipment. Critique and concerns are also valued as I'm making this up as I go. 
Most of the mobs currently coded (About 15) are inspired by people's off-handed suggestions, so please throw it at me. 

[wip]

**Comment by SonixApache** [comment]
BIG JOB ADDITIONS ARE FUCKING TOPPEST OF 👍 
However this post is tl;dr for now so I will give you proper feedback in a later post.

ideas i told you elsewhere:
 - baton fish / baton eel that works as a stun baton and maybe can be used as restraints or legcuffs as well?
 - goo gun fish that spits a glob of slowing liquid over an area
 - oyster that produces edible coins from time to time
 - some sorta buckler / riot shield fish thing, maybe with limited durability
 - one that gives you a temporary extra hand
 - conch for making noise
 - stamp squid for paper stamping  - make it usable inhand to swap it to pen mode, maybe?
 - For the kitchen, a fish that spits some nutritious fish oil over any food that's placed near it. Stick it near the microwaves for free food improvements!
 - ink smoke grenade that leaves residue and blinds people (either by clouding their unprotected eyes, or staining their eyewear until removed / cleaned)

**Comment by GreatBigFailure** [comment]
Figured I should go into big detail so people got what I was going for. It's a damn novel though. 
Once you get 6 hours to read it let me know what you think. For now just throw ideas for "thing a department needs, or would be useful, but doesn't have" at me.

**Comment by DrSnips** [comment]
Some sort of anomaly research points fish that research could DA or a product from a fish that could be DAed would be nice to have.

Idea for one.
* Conch-iousness Shelled Slug: acts like a posibrain allowing ghosts to join the round. Functions closer to an organic brain (can be put into debrained people, put into an mmi, etc). Once active the player in the slug *can* move around as is but is super slow as per being a slug and has some level of vocal communication.
FLAVOR INFO: The slug can gain a higher level of intelligence when exposed to other beings of higher intelligence. A simple tap on its head by someone is sufficient enough to trigger this process but it's not guaranteed to work.

**Comment by GreatBigFailure** [comment]
> Some sort of anomaly research points fish that research could DA or a product from a fish that could be DAed would be nice to have

I'm playing with the idea of a fish that just gives a moderate amount to random research types. Not too uncommon, possibly somehow salvagable for scrap. Techfin or whatever. It might step on some toes as a science "replacement" but having an inferior option you can cash in when science hasn't done research would be good.

As for the slug. I really like the idea of bringing people back into the round basically any time you can. I'll look at what needs doing with something like that, kind of depends on how hard it would be to code. It would also need a purpose if no one wants to enter the slug though. It's not like a posibrain where you make it intentionally, there is a lot of RNG to catches.

**Comment by Gazathor** [comment]
fish

**Comment by AI-Is-Rogue** [comment]
This sounds really fun, and interdepartmental stuff is always welcome

**Comment by Kammerjunk** [comment]
lol I'm not gonna read all that
sounds cool

**Comment by cloutcode** [comment]
if this doesn't come with 1. a fishing outfit so i can go full boomer mode and 2. fishing related knick knacks such as but not limited to singing wall-hung robotic fish and other cheesy bar decor, this concept MUST be destroyed from the face of this earth

**Comment by BomberBro** [comment]
There's some precedent for subjobs being very different from the normal one (IAA versus Bridge Officer, which even has different gear and access iirc) so it wouldn't be too weird to have that. 

I'll just warn you now not to get too out of scope with the job, for the first week(s?) it's added everyone is gonna play it so you'll be inundated with everything about it constantly, so that's expected. What you don't want is the job constantly shifting the focus of the round to that specific job even after the honeymoon phase, it's fine to add cool stuff you can only get from that job but be very careful that it doesn't make other job paths completely obsolete or add very strong stuff that you can easily acquire (the free e-sword from a shark sounds iffy to me because, assuming it works as a normal e-sword, it's a very strong item you'd only get from the syndicate or VERY rarely from vaults).

**Comment by GreatBigFailure** [comment]
> if this doesn't come with 1. a fishing outfit so i can go full boomer mode and 2. fishing related knick knacks such as but not limited to singing wall-hung robotic fish and other cheesy bar decor, this concept MUST be destroyed from the face of this earth

I'm hoping to get anglers a rig suit akin to 20,000 leagues under the sea's valve headed look. If I can get a spriter to do them I'd be happy to make boomer fishing gear too. I might even make it boost catches a bit since a few things like being pacified and certain drinks from the bar are already planned to give small boosts.
![27c1a67cfba878814471c8456ac33da5](https://user-images.githubusercontent.com/64218965/87579979-c88f6b80-c6a4-11ea-81b0-6df0bd74b3e0.jpg)


>There's some precedent for subjobs being very different from the normal one

My main concern is not having angler eat shaft miner slots. I play a lot of mechanic and that job is 100% based around having materials. I hate not having shaft miners, ruins my round. Since angler is worse at getting materials than miner I'd much prefer the ability to have both. 
On top of that it just lets me be more freeform with the design. Like I would probably add a "vault fisher" sub-job to angler, etc. 
Making it a miner sub-job is definitely still a likely option though. Partially because I dread adding angler lockers everywhere. 
I do fully admit I kind of just want to say I made a new job too.

> job paths completely obsolete or add very strong stuff that you can easily acquire (the free e-sword from a shark sounds iffy to me because, assuming it works as a normal e-sword

Trust me I'm focused on avoiding this. I've had to turn down a lot of suggestions because I don't want angler to suddenly be supplying medical with fish-themed surgery tools and leaving science in the rain. There will definitely need to be tweaks made but that'll come with proper testing, initial PRs, and the first week or two after merge. 
As for the fish-e-sword. It's balanced against botany's deathnettle. By default it's a lot weaker than a normal esword and I might add a use limit to it or something. If you get really lucky with mutations and use autism bait it could outshine an esword but only with big effort.

**Comment by cloutcode** [comment]
>deep sea diver suit

big penis activated

**Comment by DrSnips** [comment]
We already have "survivor rig suits" which look like diving suits more than space suits if you ask me. (Full disclosure: I really don't like the survivor suits not because they are bad sprites I just dislike their flavor/design because their look doesn't feel right as a space suit.)

**Comment by Eneocho** [comment]
Did you add the Bluespace sea thing to the Bluespace fish as I recommended on the thread?

**Comment by GreatBigFailure** [comment]
>Did you add the Bluespace sea thing to the Bluespace fish as I recommended on the thread?

Supermatter themed fish is on hold for a bit. It'll still get made but I think the mob list was getting over ambitious so I'm moving on to the other lists for now until those are also fleshed out. Then more can be made. I still really like the idea though, it's happening at some point.

**Comment by Theophrastus1** [comment]
fucking stole my metroid jelly idea for xenobio

**Comment by GreatBigFailure** [comment]
>fucking stole my metroid jelly idea for xenobio

Your pomfcoin royalties are in the mail

**Comment by Hatterhat** [comment]
idea: a fish that flails around and throws people in melee range, that you can eventually pick up and use to hit people, which flings them

name it the smackerel

**Comment by Theophrastus1** [comment]
> 
> 
> idea: a fish that flails around and throws people in melee range, that you can eventually pick up and use to hit people, which flings them
> 
> name it the smackerel

or give it some kind of item that binds two items together so things like stunchucks can exist

**Comment by west3436** [comment]
Can reopen this and #27984 if @GreatBigFailure ever picks this back up.
